umbraco/Umbraco-CMS
# V11: Add missing AdditionalData mapping

# Notes
- Removed private setter from AdditionalData so we can map it
- Added mapping for AdditionalData

# How to test
- I Used a SendingContentNoticiationHandler to add some values to additionalData.
- Create a document type with allow as root set to true
- Create a content for it
- Open dev tools and go to network tab
- Refresh the page
- Find the GetById request
- The metadata object should now be filled with your data like so:
 
![image](https://user-images.githubusercontent.com/70372949/207062340-1c09e9af-238b-4e07-a147-f5367d50b07b.png)


Code snippet: 
```
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.Events;
using Umbraco.Cms.Core.Notifications;

namespace Umbraco.Cms.Web.UI;
public class PreviewUrlSendingContentNotificationHandler : INotificationAsyncHandler<SendingContentNotification>
{
    public PreviewUrlSendingContentNotificationHandler()
    {
    }
    public async Task HandleAsync(SendingContentNotification notification, CancellationToken cancellationToken)
    {
        notification.Content.AdditionalData["someValue"] = "HelloWorld";
        notification.Content.AdditionalData["someObjectValue"] = new {Url = "http://google.com"};
        notification.Content.AdditionalData["myList"] = new List<string>() {"Dog", "Cat"};
    }
}

public class MyComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder) => builder
        .AddNotificationAsyncHandler<SendingContentNotification, PreviewUrlSendingContentNotificationHandler>();
}

```
